FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SNTPClientTask.c

```diff
@@ -0,0 +1,723 @@
+ * interval between consecutive system clock time queries in the sample application task.
+ */
+#define CLOCK_QUERY_TASK_DELAY_MS           ( 1000 )
+
+/**
+ * @brief The constant for storing the number of milliseconds per FreeRTOS tick in the system.
+ * @note This value represents the time duration per tick from the perspective of the
+ * of Windows Simulator based FreeRTOS system that carries lagging clock drift in relation to
+ * internet time or UTC time. Thus, the actual time duration value per tick of the system will be
+ * larger from the perspective of internet time.
+ */
+#define MILLISECONDS_PER_TICK               ( 1000 / configTICK_RATE_HZ )
+
+/**
+ * @brief Utility macro to convert the fractions part of SNTP timestamp to milliseconds.
+ */
+#define SNTP_FRACTIONS_TO_MILLISECONDS( fractions )    ( fractions / ( 1000 * SNTP_FRACTION_VALUE_PER_MICROSECOND ) )
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The definition of the @ref NetworkContext_t structure for the demo.
+ * The structure wraps a FreeRTOS+TCP socket that is used for UDP communication
+ * with time servers.
+ *
+ * @note The context is used in the @ref UdpTransportInterface_t interface required
+ * by the coreSNTP library.
+ *
+ */
+struct NetworkContext
+{
+    Socket_t xSocket;
+};
+
+/**
+ * @brief Structure aggregating state variables for RAM-based wall-clock time
+ * in Coordinated Universal Time (UTC) for system.
+ *
+ * @note This demo uses the following mathematical model to represent current
+ * time in RAM.
+ *
+ *  Total Time Elapsed since last SNTP synchronization =
+ *      No. of FreeRTOS ticks since last time synchronization *
+ *      Time Duration between consecutive ticks
+ *
+ *  Slew Adjustment = (Slew Rate * Total Time Elapsed Since Last Tine Synchronization)
+ *
+ *  Current Time = Base Time +
+ *                 Time Elapsed Since Last synchronization +
+ *                 Slew Adjustment
+ */
+typedef struct SystemClock
+{
+    UTCTime_t baseTime;
+    TickType_t lastSyncTickCount;
+    uint32_t pollPeriod;
+    uint32_t slewRate; /* Seconds/Seconds */
+    bool firstTimeSyncDone;
+} SystemClock_t;
+
+/**
+ * @brief Shared global variables for representing UTC/wall-clock time in system.
+ */
+static SystemClock_t systemClock;
+
+/**
+ * @brief Global variables of mutex protecting access to the shared memory of the
+ * system clock parameters.
+ */
+static SemaphoreHandle_t xMutex = NULL;
+static StaticSemaphore_t xSemaphoreMutex;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Utility function to convert the year to UNIX time representation of seconds
+ * since 1st Jan 1970 00h:00m:00s seconds.
+ * This utility does account for leap years.
+ *
+ * @note The time represented by the passed year is the
+ * first second on 1st January of the year.
+ *
+ * @param[in] The year to translate.
+ */
```

Comment:
```suggestion
 * @brief Utility function to convert the year to UNIX time representation of seconds
 * i.e. number of seconds since 1st Jan 1970 00h:00m:00s.
 *
 * This utility does account for leap years.
 *
 * @param[in] The year to translate.
 */
```